File: decl-py/tests/internal/conformance_test.py
"""conformance (tests/internal/checks.json): the fixture judge and the
corpus walk it rests on."""

from __future__ import annotations

from pathlib import Path

from decl.conformance import judge_fixture, walk_decl

ROOT = Path(__file__).resolve().parents[3]


def test_judge() -> None:
    files = list(walk_decl(str(ROOT / "tests/modules")))
    assert len(files) == 11 and files == sorted(files)
    assert any(f.endswith("tests/modules/basic/main.decl") for f in files)
    valid = judge_fixture(str(ROOT / "tests/validation/types/valid/predicates.decl"), True)
    wrong = judge_fixture(str(ROOT / "tests/validation/types/invalid/empty_range.decl"), True)
    right = judge_fixture(str(ROOT / "tests/validation/types/invalid/empty_range.decl"), False)
    assert valid["ok"], valid["detail"]
    assert not wrong["ok"] and "E4011" in wrong["detail"], wrong["detail"]
    assert right["ok"], right["detail"]

File: decl-ts/tests/internal/conformance_test.ts
// conformance (tests/internal/checks.json): the fixture judge and the
// corpus walk it rests on.
import { join } from 'node:path';
import { initParser } from '../../src/node.ts';
import { walkDecl, judgeFixture } from '../../src/conformance.ts';
import { check, total, root } from '../common/check.ts';

await initParser();
const files = [...walkDecl(join(root, 'tests/modules'))];
const sorted = files.every((f, i) => i === 0 || files[i - 1] < f);
const valid = judgeFixture(join(root, 'tests/validation/types/valid/predicates.decl'), true);
const wrong = judgeFixture(join(root, 'tests/validation/types/invalid/empty_range.decl'), true);
const right = judgeFixture(join(root, 'tests/validation/types/invalid/empty_range.decl'), false);
check(
  'judge',
  files.length === 11 &&
    sorted &&
    files.some((f) => f.endsWith('tests/modules/basic/main.decl')) &&
    valid.ok &&
    !wrong.ok &&
    wrong.detail.includes('E4011') &&
    right.ok,
  JSON.stringify({ n: files.length, sorted, valid, wrong, right }),
);
total();
